core-libs/setup/ssr/error-handling/server-error-response-handlers/server-error-response-handlers.ts

```diff
@@ -11,36 +11,25 @@ import {
 } from '../server-error-response/cx-server-error-response';
 
 /**
- * Handles CmsPageNotFoundServerErrorResponse which results in a fallback to client side rendering.
- * The document content is sent to the client with the appropriate 404 status code.
+ * Returns default handlers which results in a fallback to client side rendering.
+ * - If CmsPageNotFOundServerErrorResponse occurred, the document content is sent to the client with the appropriate 404 status code.
+ * - If UnknownServerErrorResponse occurred, the document content is sent to the client with the appropriate status 500 code.
  *
  * @param documentContent The document content to be sent to the client.
  * @returns The error request handler.
  */
-export const handleCmsPageNotFoundErrorResponse =
-  (documentContent: string): ErrorRequestHandler =>
-  (err, _req, res, next) => {
-    if (!res.headersSent && err instanceof CmsPageNotFoundServerErrorResponse) {
-      res.set('Cache-Control', 'no-store');
-      res.status(404).send(documentContent);
-    } else {
-      next(err);
-    }
-  };
-
-/**
- * Handles UnknownServerErrorResponse which results in a fallback to client side rendering.
- * The document content is sent to the client with the appropriate status 500 code.
- *
- * @param documentContent The document content to be sent to the client.
- * @returns The error request handler.
- */
-export const handleUnknownServerErrorResponse =
+export const defaultServerErrorResponseHandlers =
   (documentContent: string): ErrorRequestHandler =>
   (err, _req, res, next) => {
     if (!res.headersSent && err instanceof UnknownServerErrorResponse) {
       res.set('Cache-Control', 'no-store');
       res.status(500).send(documentContent);
+    } else if (
+      !res.headersSent &&
+      err instanceof CmsPageNotFoundServerErrorResponse
+    ) {
+      res.set('Cache-Control', 'no-store');
+      res.status(404).send(documentContent);
     } else {
       next(err);
     }
```

Comment:
I apologize I might not have been clear in my last comment.
My intention was to:
- leave 2 separate handlers in our public API (in case customers want to pick and mix)
- and **moreover** expose one "wrapper" handler which composes the other two - that is ergonomic to use and add new stuff underneeth in the future
- via schematics install this "wrapper" error handler

Would you agree?

Reply:
I was thinking about such a solution, but I haven't found a nice way to return two handlers of type ErrorRequestHandler return in one `defaultServerErrorResponseHandlers` function to use it in ExpressJS API (`server.use`):

```ts
export const defaultServerErrorResponseHandlers =
  (documentContent: string): ErrorRequestHandler => //won't match the type
   getCmsPageNotFoundServerErrorResponseHandler(documentContent)
   getUnknownServerErrorResponseHandler(documentContent)
  };
 ```
  
When I extracted the body of this function to two separate functions for each case, `defaultServerErrorResponseHandlers` looked more complex + it wouldn't make much sense, because these functions and take would not be usable directly as middleware in ExpressJS.

Right now this logic can be overwritten by custom middleware (for 404, 500, or both - if we override behaviour for 404, then handling 500 still will be taken from our default handler).

Anyway, I'm open for proposals.

Reply:
Ah, you're right! My bad. Ok, let's keep it as is!

FYI, I was thinking about something like below, but only now I realized the composition of those 2 handlers doesn't work at all as I indended 🤦  Pasting it only for reference. What you have now is good!

```ts
export const handleUnknownServerError = (documentContent: string) => {
  return (err, _req, res, next) => {
    if (!res.headersSent && err instanceof UnknownServerErrorResponse) {
      res.set('Cache-Control', 'no-store');
      res.status(500).send(documentContent);
    } else {
      next(err);
    }
  };
};

export const handleCmsPageNotFoundError = (documentContent: string) => {
  return (err, _req, res, next) => {
    if (!res.headersSent && err instanceof CmsPageNotFoundServerErrorResponse) {
      res.set('Cache-Control', 'no-store');
      res.status(404).send(documentContent);
    } else {
      next(err);
    }
  };
};

export const defaultServerErrorResponseHandlers =
  (documentContent: string): ErrorRequestHandler =>
  (err, _req, res, next) => {
    handleUnknownServerError(documentContent)(err, _req, res, next);
    handleCmsPageNotFoundError(documentContent)(err, _req, res, next);
  };
```